File: aoj/volume0/n0027/Main.java
import java.util.Scanner;

public class Main {

    private static final int DAYS[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    private static final String WEEK_NAMES[] = {"Wednesday", "Thursday", "Friday", "Saturday", "Sunday", "Monday", "Tuesday"};

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        while (true) {

            int month = sc.nextInt();
            int day = sc.nextInt();

            if (month == 0) {
                break;
            }

            for (int i = 0; i < month; i++) {
                day += DAYS[i];
            }

            System.out.println(WEEK_NAMES[day % 7]);

        }

    }

}

File: aoj/volume0/n0027/Main.cpp
#include <iostream>

using namespace std;

int main() {

    const string WEEK_NAMES[] = {"Wednesday", "Thursday", "Friday", "Saturday", "Sunday", "Monday", "Tuesday"};
    const int DAYS[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

    while (1) {

        int month, day;
        cin >> month >> day;

        if (!(month || day)) {
            return 0;
        }

        for (int i = 0; i < month; i++) {
            day += DAYS[i];
        }

        cout << WEEK_NAMES[day % 7] << endl;

    }

    return 0;

}
